submittal/src/main/java/com/kyriba/submittal/api/SpeakerController.java

```diff
@@ -0,0 +1,104 @@
+package com.kyriba.submittal.api;
+
+import com.kyriba.submittal.domain.dto.PersonalInfoDto;
+import com.kyriba.submittal.domain.dto.SpeakerDataDto;
+import com.kyriba.submittal.domain.dto.TopicDto;
+import com.kyriba.submittal.service.SpeakerService;
+import lombok.Builder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.SessionAttribute;
+
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+
+
+/**
+ * @author M-ABL
+ */
+// TODO: setup security. Let's assume a filter retrieves a speaker's id by email/password and puts the id to the session.
+@RestController
+@RequestMapping(value = "/api/v1/speaker",
+    consumes = APPLICATION_JSON_UTF8_VALUE,
+    produces = APPLICATION_JSON_UTF8_VALUE)
+public class SpeakerController
+{
+  private final SpeakerService speakerService;
+
+
+  @Autowired
+  SpeakerController(final SpeakerService speakerService)
+  {
+    this.speakerService = speakerService;
+  }
+
+
+  @PostMapping(path = "/register")
+  public SpeakerResponse register(@RequestBody final SpeakerRequest speaker)

```

Comment:
public and final are useless here

Reply:
I always use 'final' for arguments

Reply:
Controllers are storage for meta-information, so all unnecessary keywords may be removed for simplification